atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBackupRunner.java

```diff
@@ -0,0 +1,204 @@
+
+        Preconditions.checkState(boundReadability != BoundReadability.BOTH,
+                "We had both backup and active bounds readable! This is unexpected. Please contact support.");
+        Preconditions.checkState(boundReadability != BoundReadability.NEITHER,
+                "We had an unreadable active bound with no backup! This is unexpected. Please contact support.");
+        return boundReadability;
+    }
+
+    private BoundReadability getReadability(BoundData boundData) {
+        boolean boundReadable = boundData.bound() == null
+                || CassandraTimestampUtils.isValidTimestampData(boundData.bound());
+        boolean backupBoundReadable = CassandraTimestampUtils.isValidTimestampData(boundData.backupBound());
+
+        if (boundReadable) {
+            return backupBoundReadable ? BoundReadability.BOTH : BoundReadability.BOUND;
+        }
+        return backupBoundReadable ? BoundReadability.BACKUP : BoundReadability.NEITHER;
+    }
+
+    private BoundData getCurrentBoundData(Cassandra.Client client) {
+        checkTimestampTableExists();
+
+        ByteBuffer selectQuery = CassandraTimestampUtils.constructSelectFromTimestampTableQuery();
+        CqlResult existingData = executeQueryUnchecked(client, selectQuery);
+        Map<String, byte[]> columnarResults = CassandraTimestampUtils.getValuesFromSelectionResult(existingData);
+
+        return ImmutableBoundData.builder()
+                .bound(columnarResults.get(CassandraTimestampUtils.ROW_AND_COLUMN_NAME))
+                .backupBound(columnarResults.get(CassandraTimestampUtils.BACKUP_COLUMN_NAME))
+                .build();
+    }
+
+    private void checkTimestampTableExists() {
+        CassandraTables cassandraTables = cassandraKeyValueService.getCassandraTables();
+        Preconditions.checkState(
+                cassandraTables.getExisting().contains(AtlasDbConstants.TIMESTAMP_TABLE.getQualifiedName()),
+                "[BACKUP/RESTORE] Tried to get bound data when the timestamp table didn't exist!");
+    }
+
+    private void executeAndVerifyCas(Cassandra.Client client, Map<String, Pair<byte[], byte[]>> casMap) {
+        ByteBuffer casQueryBuffer = CassandraTimestampUtils.constructCheckAndSetMultipleQuery(casMap);
+
+        CqlResult casResult = executeQueryUnchecked(client, casQueryBuffer);
+        CassandraTimestampUtils.verifyCompatible(casResult, casMap);
+    }
+
+    private CqlResult executeQueryUnchecked(Cassandra.Client client, ByteBuffer query) {
+        try {
+            return queryRunner().run(client,
+                    AtlasDbConstants.TIMESTAMP_TABLE,
+                    () -> client.execute_cql3_query(query, Compression.NONE, ConsistencyLevel.QUORUM));
+        } catch (TException e) {
+            throw Throwables.rewrapAndThrowUncheckedException(e);
+        }
+    }
+
+    private CassandraClientPool clientPool() {
+        return cassandraKeyValueService.clientPool;
```

Comment:
This should be wrapped in a getter - make CKVS.clientPool protected and add `clientPool()`.